salt/modules/win_system.py

```diff
@@ -792,14 +793,44 @@ def get_domain_workgroup():
             return {'Workgroup': computer.Domain}
 
 
+def _try_parse_datetime(time_str, fmts):
+    '''
+    Attempts to parse the input time_str as a date.
+
+    :param str time_str: A string representing the time
+    :param list fmts: A list of date format strings
+
+    :return: Returns a datetime object if parsed properly. Otherwise None
+    :rtype datetime
+    '''
+    result = None
+    for fmt in fmts:
+        try:
+            result = datetime.strptime(time_str, fmt)
+            break
+        except ValueError:
+            pass
+    return result
+
+
 def get_system_time():
     '''
     Get the system time.
 
-    :return: Returns the system time in HH:MM AM/PM format.
+    :return: Returns the system time in HH:MM:SS AM/PM format.
     :rtype: str
     '''
-    return datetime.strftime(datetime.now(), "%I:%M %p")
```

Comment:
There isn’t an issue with strftime(), but there is one with datetime.now(). On Windows, datetime.now() won’t return the correct time if the timezone has been changed within the current python process.

From: Mike Place [mailto:notifications@github.com]
Sent: Tuesday, July 19, 2016 4:00 PM
To: saltstack/salt <salt@noreply.github.com>
Cc: Molly Kreis <molly.kreis@ni.com>; Author <author@noreply.github.com>
Subject: Re: [saltstack/salt] Dev/mkreis/win system fixes (#34788)


In salt/modules/win_system.py<https://github.com/saltstack/salt/pull/34788#discussion_r71422313>:

>      :rtype: str

>      '''

> -    return datetime.strftime(datetime.now(), "%I:%M %p")

Why is this being replaced? Are you suggesting there is a bug in python's strftime()?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/saltstack/salt/pull/34788/files/c3a083dfa36c294de17ef6d064c766508965cad6#r71422313>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ATl0XIgohinU1jIBzT_Z7QDCcktzpQ1jks5qXTrTgaJpZM4JP9Tb>.